python/lsst/multiprofit/plotting/plot_model_rgb.py

```diff
@@ -0,0 +1,398 @@
+            model_band = models[band]
+            sigma_inv = observation.sigma_inv.data
+            sigma_inv_good = sigma_inv > 0
+            variance_band = np.empty_like(sigma_inv)
+            variance_band[sigma_inv_good] = sigma_inv[sigma_inv_good] ** -2
+            variance_band[~sigma_inv_good] = np.nan
+            if plot_chi_hist:
+                chi_good = (sigma_inv > 0) & np.isfinite(sigma_inv)
+                chi_unweighted = (observation.image.data[chi_good] - model_band[chi_good]) * sigma_inv[
+                    chi_good
+                ]
+                chis_unweighted[band] = chi_unweighted
+        weight_channel_new = weights_channel[idx_band]
+        idx_channel_new = int(weight_channel_new // 1)
+        if idx_channel_new == idx_channel:
+            weight_low = weight_channel_new - weight_channel
+            weight_high = 0.0
+        else:
+            weight_low = idx_channel_new - weight_channel
+            weight_high = weight_channel_new - idx_channel_new
+        assert weight_high >= 0
+        assert weight_low >= 0
+        if weight_low > 0:
+            data_band = observation.image.data * weight_low
+            add_if_not_none(images_data, idx_channel, data_band * weight)
+            add_if_not_none(masks_inv_rgb, idx_channel, observation.mask_inv.data * weight_low)
+            if has_model:
+                add_if_not_none(images_data_unweighted, idx_channel, data_band)
+                model_sub = model_band * weight_low
+                add_if_not_none(images_model, idx_channel, model_sub * weight)
+                add_if_not_none(images_model_unweighted, idx_channel, model_sub)
+                add_if_not_none(images_sigma_inv, idx_channel, variance_band * weight_low)
+        if (idx_channel_new != idx_channel) and (weight_high > 0):
+            data_band = observation.image.data * weight_high
+            images_data[idx_channel_new] = data_band * weight
+            masks_inv_rgb[idx_channel_new] = observation.mask_inv.data * weight_low
+            if has_model:
+                images_model_unweighted[idx_channel_new] = data_band
+                model_sub = model_band * weight_high
+                images_model[idx_channel_new] = model_sub * weight
+                images_model_unweighted[idx_channel_new] = model_sub
+                images_sigma_inv[idx_channel_new] = variance_band * weight_high
+        weight_channel = weight_channel_new
+        idx_channel = idx_channel_new
+
+    # convert variance to 1/sigma
+    if has_model:
+        for idx in range(3):
+            images_sigma_inv[idx] = 1 / np.sqrt(images_sigma_inv[idx])
+
+    if rgb_min_auto or rgb_stretch_auto:
+        # The model won't have negative pixels, so it ought to stretch fine
+        # the max/stretch is not as important anyway
+        rgb_min, rgb_max = np.nanpercentile(
+            np.concatenate([image[mask_inv != 0] for mask_inv, image in zip(masks_inv_rgb, images_data)]),
+            (5, 95),
```

Comment:
It might be useful to make these limits configurable (and other percentile units defined in this function).

Reply:
I agree but I think I will defer this until I get around to trying Nate's new RGB image code (no ticket yet, I have enough on my plate).